frontend/packages/employee-frontend/src/components/messages/MessagesPage.tsx

```diff
@@ -54,22 +56,34 @@ export default React.memo(function MessagesPage() {
     if (unit) loadGroups(unit.id)
   }, [unit])
 
-  const loadDraftMessages = useRestApi(getDraftBulletins, setDraftMessages)
-  useEffect(() => {
-    if (unit) loadDraftMessages(unit.id)
-  }, [unit])
+  const setMessageCountResult = useCallback(
+    (result: Result<Record<MessageBoxType, number>>) =>
+      result.isSuccess ? setMessageCounts(result.value) : undefined,
+    [setMessageCounts]
+  )
+  const loadMessageCounts = useRestApi(
+    (unitId: string) =>
+      Promise.all([
+        getDraftBulletins(unitId, 1, 1),
+        getSentBulletins(unitId, 1, 1)
+      ]).then(([drafts, sent]) =>
+        drafts.chain((ds) =>
+          sent.map((ss) => ({ SENT: ss.total, DRAFT: ds.total }))
+        )
+      ),
```

Comment:
This feels somewhat hard to understand to me, but I don't really have better suggestions either. Part of it might be that Array.map vs Result.map always confuses me (maybe it should be e.g. Result.transform?) and the word sent being both singular and plural does not help either. :D

Reply:
Well `map` is really not specific to arrays or lists or other iterables, it's a property of [functors](https://en.wikipedia.org/wiki/Functor_(functional_programming)). Same thing with `chain` (sometimes known as `flatMap`).